### PR TITLE
Add even more page objects

### LINKED ISSUE #715: Rewrite integration tests for donation-goal-edit component to use page objects
Falls under umbrella issue #697.

## Resources

- What's a page object? [[1](http://martinfowler.com/bliki/PageObject.html),[2](https://wyeworks.com/blog/2015/5/13/using-the-page-object-pattern-with-ember-cli/)]
- [Quickstart guide](http://ember-cli-page-object.js.org/docs/v1.6.x/quickstart)
- See [`tests/pages/components/donations/create-donation.js`](https://github.com/code-corps/code-corps-ember/blob/develop/tests/pages/components/donations/create-donation.js) and [`tests/integration/components/donations/create-donation-test.js`](https://github.com/code-corps/code-corps-ember/blob/develop/tests/pages/components/donations/create-donation.js) for in-house examples of, respectively, a page object and the page object being utilized in an integration test.
- To generate a page object, run `ember generate page-object component/<component-name>` in the project directory.

### PR DESCRIPTION
**Pull request by JoshSmith**
# What's in this PR?

Fixes #715 #723 #974

### CODE REVIEW

File: tests/integration/components/donation-goal-edit-test.js [hunk 1]
```diff
@@ -14,36 +18,32 @@ function setHandlers(context, { saveHandler = function() {}, cancelHandler = fun
...
-  this.render(hbs`{{donation-goal-edit donationGoal=donationGoal cancel=cancelHandler save=saveHandler}}`);
-
-  assert.equal(this.$('.donation-goal-edit').length, 1);
-});
-
 test('it renders cancel button, when canCancel is true', function(assert) {
   assert.expect(1);
 
-  this.render(hbs`{{donation-goal-edit donationGoal=donationGoal canCancel=true cancel=cancelHandler save=saveHandler}}`);
+  page.render(hbs`{{donation-goal-edit donationGoal=donationGoal canCancel=true cancel=cancelHandler save=saveHandler}}`);
```
Comment (snewcomer) [inline tests/integration/components/donation-goal-edit-test.js]: Haven't seen this use before (`page.render`).  Guessing you found this in the documentation?

Comment (JoshSmith) [inline tests/integration/components/donation-goal-edit-test.js]: @snewcomer it's because of the `ember-cli-page-object` page object rendering.

Comment (JoshSmith) [inline tests/integration/components/donation-goal-edit-test.js]: In `beforeEach` you do `page.setContext(this);`

**Comment by JoshSmith** [comment]
Thanks @snewcomer!

**Pull request merged**